imu_driver/imu_controller.py

```diff
@@ -0,0 +1,188 @@
+import sys
+import os
+import logging
+import numpy as np
+from datetime import datetime, timedelta, timezone
+
+sys.path.insert(0, os.path.abspath(os.path.join(os.path.dirname(__file__), '..')))
+from imu_driver.adafruit_lsm6ds.lsm6ds33 import LSM6DS33
+from imu_driver.adafruit_lsm6ds import Rate, AccelRange, GyroRange
+import RPi.GPIO as GPIO
+
+logger = logging.getLogger(__name__)  
+logger.setLevel(logging.DEBUG)
+
+def parse_fifo_data(data: bytearray, signed: bool=True) -> None:
+    """Parse FIFO data from bytes to decimal."""
+    return int.from_bytes(data, byteorder='little', signed=signed)
+
+class ImuController():
+    """
+    IMU Controller connects to IMU and handles reading the FIFO
+    ...
+    Attributes
+    ----------
+    i2c_handler : object
+        I2C handler
+
+    Methods
+    -------
+    _init_imu(fifo_theshold)
+        Initialize IMU with FIFO configuration
+    _imu_interrupt_handler()
+        Handle FIFO threshold interrupt, process FIFO data
+    _process_fifo_data(data_buffer)
+        Process byte array into sensor readings
+    """
+    IMU_ADDRESS = 0x6a
+    IMU_INT1_PIN = 14
+    WORDS_PER_PATTERN = 18
+    TIMESTAMP_INCREMENT = 25 # microseconds
+    TIMESTAMP_RESET_THRESHOLD = 1000000
+    FIFO_THRESHOLD = 180
+    def __init__(self, i2c_handler):
+        self._i2c_handler = i2c_handler
+        self._init_imu(self.FIFO_THRESHOLD)
+
+        # Setup INT1 Pin
+        GPIO.setup(self.IMU_INT1_PIN, GPIO.IN)
+        GPIO.add_event_detect(self.IMU_INT1_PIN, GPIO.RISING, callback=self._imu_interrupt_handler) 
+
+    def _init_imu(self, fifo_threshold: int=180):
```

Comment:
Add spaces around `=` to make type hinting cleaner
```suggestion
    def _init_imu(self, fifo_threshold: int = 180):
```